### DISCUSSION TITLE
Mixing Chai and Jest

### DISCUSSION THREAD

**Discussion by janniks** [vitest-dev/vitest#4110] [Q&A]
I noticed it's not encouraged to mix the both APIs: https://github.com/vitest-dev/vitest/issues/2843#issuecomment-1424352509
Is that still the recommendation? @sheremet-va

In my specific case I'm not able to figure out how to use `expect.extend({ ... })` but also `chai.use`.
The _extend_ is working fine. But I need to also register an AsymmetricMatcher (does this work with extend?). The chai plugin looks ok, but then the matched is just not defined. 😬

```ts
expect.extend({
  // normal custom matchers
})

// chai plugin for asymmetric matcher (inspired by vitest source of ArrayContaining)
chai.use((chai, utils) => {
  utils.addMethod(
    chai.expect,
    "listContaining",
    (expected: string) => new ListContaining(expected)
  );

  (chai.expect as any).not = {
    ListContaining: (expected: string) => new ListContaining(expected, true),
  };
});
```

**Comment by sheremet-va** [comment]
I'd recommend using `expect.extend`. It does register an async matcher.

**Reply by janniks** [reply]
Interesting thanks, I'll try again to reproduce. I was basically trying to create an alternative to arrayContaining (but that accesses a sub-property of an object, rather than working on an array directly (due to a project using a DSL with wrapper objects)
